Source/core/CMakeLists.txt

```diff
@@ -217,8 +217,10 @@ if(BUILD_TESTS)
 endif()
 
 target_link_libraries(${TARGET}
+
+        PUBLIC
+            CompileSettings::CompileSettings
```

Comment:
Add PRIVATE CompileSettings to Thunder/CMakeLists.txt instead of exporting compilesettings from Core.